src/main/java/software/sigma/bu003/internship/coparts/security/service/CustomOidcUserService.java

```diff
@@ -0,0 +1,86 @@
+package software.sigma.bu003.internship.coparts.security.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.core.OAuth2AuthenticationException;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+import software.sigma.bu003.internship.coparts.security.model.GoogleUserInfo;
+import software.sigma.bu003.internship.coparts.security.model.User;
+import software.sigma.bu003.internship.coparts.security.repository.UserRepository;
+
+import java.time.LocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+public class CustomOidcUserService extends OidcUserService {
+
+    private final UserRepository userRepository;
+
+    @Override
+    public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2AuthenticationException {
+        OidcUser oidcUser = super.loadUser(userRequest);
+
+        return takeOidcUser(oidcUser);
+    }
+
+    private OidcUser takeOidcUser(OidcUser oidcUser) {
+        GoogleUserInfo googleUserInfo = new GoogleUserInfo(oidcUser.getAttributes());
+
+        saveUserInfo(googleUserInfo);
+
+        return oidcUser;
+    }
+
+    private void saveUserInfo(GoogleUserInfo googleUserInfo){
+        User user = userRepository.findByEmail(googleUserInfo.getEmail())
+                .map(userFromDB -> updateUser(userFromDB, googleUserInfo))
+                .orElseGet(() -> createNewUser(googleUserInfo));
+
+        userRepository.save(user);
+    }
+
+    private User createNewUser(GoogleUserInfo googleUserInfo){
+        return User.builder()
+                .googleId(googleUserInfo.getGoogleId())
+                .email(googleUserInfo.getEmail())
+                .fullName(googleUserInfo.getFullName())
+                .givenName(googleUserInfo.getGivenName())
+                .familyName(googleUserInfo.getFamilyName())
+                .imageUrl(googleUserInfo.getPictureUrl())
+                .locale(googleUserInfo.getLocale())
+                .lastVisit(LocalDateTime.now())
+                .build();
+    }
+
+    private User updateUser(User userFromDB, GoogleUserInfo googleUserInfo) {
+        if(userFromDB.getGoogleId() == null ||
```

Comment:
may be?
![image](https://user-images.githubusercontent.com/13138548/207897749-cd162318-b4b3-4bbc-99de-f938dbf1b5cf.png)
also a few questions more 
1. userFromDB can be null or not?
2. Looks like every time you connect you sync this user?